Add new "compose_process_api" as a wrapper for "compose_process"

tenzap opened the pull request:
* When calling "compose_process" from a form within a browser there were 2 issues:
   - After logging-in the POSTed data was lost and thus nothing was sent
   - If POSTing while already logged-in, the return message was displayed
     on the same URL, what could lead to a re-submission if the user
     refreshed the page.

 * This creates a wrapper method calling compose_process that will fix it

 * Now, when posting to /index.php/messages/compose_process_api before the user
   is logged-in, this will redirect the user to the URL containing the form
   he posted.
    - If in the process the POSTed data is lost, the user gets redirected by
      using the window.history.go function instead of the referrer
      so that the data of the form is not lost.
    - If the POSTed data is kept, then it is automatically submitted

 * After submission, the user is redirected to a new page showing the form result.

tenzap commented:
All the new code in `if(!$_POST)` can also be moved to `compose_process` method instead.
Doing so would make `compose_process` benefit from the auto-rePOST after login too.

I didn't do it because I didn't know if it actually has any side-effect on the processes that already call `compose_process`.

tenzap commented:
Is it possible to discuss about this to consider a merge at some time ?

kingster commented:
>If POSTing while already logged-in, the return message was displayed on the same URL, what could lead to a re-submission if the user refreshed the page.

The browser would show a warning before resubmitting.

> All the new code in if(!$_POST) can also be moved to compose_process method instead. Doing so would make compose_process benefit from the auto-rePOST after login too.

Agree with this. 

> I didn't do it because I didn't know if it actually has any side-effect on the processes that already call compose_process.

Well, we would need to check through all scenario. I don't recall this flow much as it has been quite some time :(

tenzap commented:
Shall I put the `if(!$_POST)` changes to compose_process directly and just leave the last 2 lines in the wrapper ?
Then after merge we could wait and see for any side effect reported ?

BTW, what do you think about releasing new version soon with all the recent changes (the migration to php7/ci3 being a big change) ?

kingster commented:
Agree on first point.

Regrading next version release, lets cleanup and ensure little stability in our dev branch and we would be good to release a new version.

tenzap commented:
I moved the code for rePOSTing  to compose_process and added a redirection (the one that was in compose_process_api) on optional basis (through parameter in POST), thus compose_process_api wrapper is not needed anymore

kingster commented on application/controllers/Form_result.php:
```diff
@@ -0,0 +1,9 @@
+<?php
+class Form_result extends CI_Controller
```
Can we please change the name of this to something more meaningful, please.

tenzap replied:
What do you suggest ? I had no inspiration

tenzap replied:
This is a very general page that simply displays the message passed to it as parameter, so I gave it a general name

tenzap replied:
maybe something like "DisplayMessage" ?

kingster replied:
DisplayMessage seems better:)

tenzap replied:
Finally I chose "Info_message"
BTW, I also made the strings of Messages.php translatable

kingster commented:
Looks good now :)
